src/features/processPicker.ts

```diff
@@ -35,6 +37,111 @@ export class AttachPicker {
     }
 }
 
+export class RemoteAttachPicker {
+    public static ShowAttachEntries(args : any): Promise<string> {
+        // Grab selected name from UI
+        let name : string = args.name;
+
+        if (!name) {
+            // Config name not found. 
+            return new Promise<string>((resolve, reject) => {
+                reject(new Error("Name not defined in current configuration."));
+            });
+        }
+
+        // Build path for launch.json to find pipeTransport
+        const vscodeFolder : string = path.join(vscode.workspace.rootPath, '.vscode');
+        let launchJsonPath : string = path.join(vscodeFolder, 'launch.json');
+
+        // Read launch.json
+        let json : any = JSON.parse(fs.readFileSync(launchJsonPath).toString());
+
+        // Find correct pipeTransport via selected name
+        let config; 
+        let configIdx : number;
+        for (configIdx = 0; configIdx < json.configurations.length; ++configIdx) {
+            if (json.configurations[configIdx].name === name) {
+                config = json.configurations[configIdx];
+                break; 
+            }
+        }
+
+        if (configIdx == json.configurations.length) {
+            // Name not found in list of given configurations. 
+            return new Promise<string>((resolve, reject) => {
+                reject(new Error("Could not find configuration that matches given name"));
+            });
+        }
+
+        if (!config.pipeTransport) {
+            // Missing PipeTransport, prompt if user wanted to just do local attach.
+            return new Promise<string>((resolve, reject) => {
+                reject(new Error("Configuration \"" + args.name + "\" in launch.json does not have a " + 
+                "pipeTransport argument for pickRemoteProcess. Use pickProcess for local attach."));
+            });
+        } else {
+            let pipeCmd : string = config.pipeTransport.pipeProgram + " " + config.pipeTransport.pipeArgs.join(" ");
```

Comment:
Both pipeArgs and pipeProgram can be overwritten with the host-OS-specific options. So you need to check if those properties exist, and if so, give preference to them. See OpenDebugAD7 for an example.